x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list.tsx

```diff
@@ -94,8 +97,11 @@ export const CreateEngineButton: React.FC<{ disabled: boolean }> = ({ disabled }
     </EuiPopover>
   );
 };
+interface ListProps {
+  isCreateEngineFlyoutOpen?: boolean;
+}
 
-export const EnginesList: React.FC = () => {
+export const EnginesList: React.FC<ListProps> = ({ isCreateEngineFlyoutOpen }) => {
```

Comment:
Without this props, When create search app flyout is open ( instantiated from indices page ) doesn't redirect to engine list page after successfully creating search application. 
Adding this optional `isCreateEngineFlyoutOpen` helped with that. 

This was the best solution I could come up with. So looking for better alternative if any!